### DISCUSSION TITLE
Yolo26-DPAR

### DISCUSSION THREAD

**Discussion by cribbinsb** [ultralytics/ultralytics#24235] [Show and tell]
In case anyone find it interesting: here https://github.com/ubonpartners/yolo-dpar is work building on a set of repos including a fork of Ultralytics Yolo26 (https://github.com/ubonpartners/ultralytics) which add a few extra capabilities oriented towards improved person detection: 

- person, face, vehicle,... object detection
- 17 x body pose point (on each person detection), 5 x face keypoints (on each face detection)
- detection of 50 binary attributes on each person detection (male/female, clothing type, colour, bag,...)
- FIQA face visibility quality metric output for each face (accounting for blur/occlusion/orientation/etc)
- fused/trained Re-ID head outputting person Re-ID vectors

The interesting thing is that these are all executed simultaneously in the normal single forward pass without much overhead. Some model check points are provided that have been trained on a much larger / more varied dataset than COCO aimed at improved real world use/better generalization. The attribute training as well as being directly useful in itself works synergistically with the REID head as it greatly improves the discrimination capability of the features.

Hope you find this work of some use! Thanks to the Ultralytics team for making a base project that is so easy to work with.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @cribbinsb, thank you for sharing this interesting YOLO26-DPAR work with Ultralytics 🚀! It’s great to see a fork of Ultralytics YOLO26 being extended with multi-task capabilities like detection, pose, attributes, face quality, and Re-ID in a single forward pass. This is an automated response to help get the discussion started, and an Ultralytics engineer will also assist soon 😊

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

For a project showcase like this, feel free to share any additional details that may help the community evaluate and reproduce results, such as example outputs, benchmark comparisons, training setup, or usage notes 📊

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by glenn-jocher** [reply]
Thanks for sharing this here — the single-pass combination of detection, pose, attributes, FIQA, and Re-ID is interesting, and adding a short benchmark plus a minimal usage example would make it easier for others to compare it with the standard [pose workflow](https://docs.ultralytics.com/tasks/pose/) and [BoT-SORT Re-ID integration](https://docs.ultralytics.com/reference/trackers/bot_sort/).

**Reply by cribbinsb** [reply]
Hi @glenn-jocher there is already a reasonably simple example here https://github.com/ubonpartners/yolo-dpar/blob/main/yolo-dpa-test.py which displays detections/pose/face/face points on live video and when you click on a person shows the live detected attributes. In terms of benchmark what were you thinking - like inference time?

**Reply by glenn-jocher** [reply]
Yes—mainly end-to-end latency or FPS on fixed hardware, with `imgsz`, precision, batch size, and model size noted, plus a few held-out quality metrics for detection, pose, attributes, and Re-ID; a small side-by-side table using Ultralytics [Benchmark mode](https://docs.ultralytics.com/modes/) as the speed baseline would make the comparison much easier for others to evaluate.

**Reply by cribbinsb** [reply]
Hopefully this is along the lines of what you were thinking of; this was captured using the built-in benchmark mode. The val dataset used is 16K images from 11 other val datasets but mainly a mix of coco, openimages, objects365. It was measured on RTX5090 using fp16 tensorrt, seems in this case the latency overhead is around 10% say comparing yolo26l.pt baseline to yolo26l-e2e-v10r-080426.pt. The later includes all the face,pose,attributes and trained REID head. 

<img width="3845" height="1805" alt="combined_results_table" src="https://github.com/user-attachments/assets/4d0aadc7-a552-46f8-90fa-9d1086d42ae5" />

**Reply by glenn-jocher** [reply]
Yes, that’s the kind of comparison I meant; the ~10% `fp16` TensorRT overhead on fixed `RTX 5090` hardware is useful context, and if you can paste the key numbers as text with `imgsz`, batch, exact model/engine names, and the `benchmark` invocation, it’ll be much easier for others to search, reproduce, and compare—task metrics on that same 16K-image set would round it out nicely.
